servers/cu/src/domain/client/ao-process.js

```diff
@@ -1466,6 +1593,48 @@ export function saveCheckpointWith ({
               return { id: res.id, encoding: 'gzip' }
             }
           )
+          .bichain(
+            (ctx) => {
+              /**
+               * If file checkpoints are disabled, skip this step.
+               */
+              if (DISABLE_PROCESS_FILE_CHECKPOINT_CREATION) return Rejected(ctx)
+              console.log(100, { ctx })
+              /**
+               * If the upload to arweave fails, we will save a backup checkpoint in the file system.
+               * Then, we will add it to the sqlite directory.
+               */
+              const evaluation = {
+                processId,
+                moduleId,
+                timestamp,
+                nonce,
+                ordinate,
+                epoch,
+                blockHeight,
+                cron,
+                encoding: 'gzip'
+              }
+              if (Memory) {
+                writeProcessMemoryFile({
+                  Memory,
+                  evaluation
+                })
+                  .then((path) => {
+                    writeFileRecord(evaluation, path)
+                  })
```

Comment:
Should probably be `.then((path) => writeFileRecord(evaluation, path))`. Otherwise, the `Promise` is left dangling and won't trigger the `catch` and instead bubble.